src/bridge/StellarJSBridge.js

```diff
@@ -0,0 +1,349 @@
+import { parseAPIValue, getServer, formatAPICurrencyXLM } from 'api/Stellar'
+import { Transaction as StrTransaction, Asset } from 'stellar-sdk'
+import { StrKey } from 'stellar-base'
+import AdvancedOptionsStellarKind from 'components/AdvancedOptions/StellarKind'
+import { getAccountPlaceholderName } from 'helpers/accountName'
+import { NotEnoughBalance } from 'config/errors'
+import type { WalletBridge, EditProps } from './types'
+
+type Transaction = {
+  amount: BigNumber,
+  recipient: string,
+  fee: BigNumber,
+  memoError?: ?string,
+  memoType?: { label: string, value: ?string },
+  memo?: ?string,
+}
+
+const paymentsToOperation = async (id, payments, currency, account) => {
+  const parsedPayments = []
+  for (const payment of payments) {
+    let from
+    let to
+    let value
+
+    if (payment.type === 'create_account') {
+      from = payment.funder
+      to = payment.account
+      value = parseAPIValue(payment.starting_balance)
+    } else {
+      from = payment.from
+      to = payment.to
+      value = parseAPIValue(payment.amount)
+    }
+
+    const operation: Operation = {
+      id: payment.id,
+      hash: payment.transaction_hash,
+      accountId: id,
+      type: account === from ? 'OUT' : 'IN',
+      value,
+      fee: new BigNumber(100),
+      blockHash: payment.transaction_hash,
+      blockHeight: +payment.paging_token,
+      senders: [from],
+      recipients: [to],
+      date: new Date(payment.created_at),
+      transactionSequenceNumber: payment.transaction_hash,
+    }
+
+    if (payment.type === 'create_account') {
+      operation.senders = [payment.funder]
+      operation.recipients = [payment.account]
+    }
+
+    parsedPayments.push(operation)
+  }
+  return parsedPayments
+}
+const getBalanceFromAccount = account => {
+  let balance = new BigNumber(0)
+
+  if (Array.isArray(account.balances) && account.balances.length > 0) {
+    const nativeBalance = account.balances.find(balance => balance.asset_type === 'native')
+    if (nativeBalance) {
+      balance = parseAPIValue(nativeBalance.balance)
+    }
+  }
+  return balance
+}
+const signAndBroadcast = async ({
+  a,
+  t,
+  deviceId,
+  isCancelled,
+  onSigned,
+  onOperationBroadcasted,
+}) => {
+  const server = getServer()
+  const amount = formatAPICurrencyXLM(t.amount)
+  const signedTransactionXDR = await signTransaction
+    .send({
+      currencyId: a.currency.id,
+      devicePath: deviceId,
+      path: a.freshAddressPath,
+      transaction: {
+        freshAddress: a.freshAddress,
+        destination: t.recipient,
+        asset: Asset.native(),
+        memo: t.memo,
+        memoType: (t.memoType && t.memoType.value) || 'MEMO_NONE',
+        amount,
+      },
+    })
+    .toPromise()
+  const signedTransaction = new StrTransaction(signedTransactionXDR)
+
+  if (!isCancelled()) {
+    onSigned()
+    const transactionResult = await server.submitTransaction(signedTransaction)
```

Comment:
Please handle timeout errors: https://www.stellar.org/developers/horizon/reference/endpoints/transactions-create.html#timeout Otherwise users may think that tx fail, but it actually succeeded.

In case of a timeout error you should inform user about it and let them try to resubmit the same transaction again (with the same seqnum) to check it's final status.